appIos/feed/feed/FeedScreen.swift

```diff
@@ -48,7 +48,7 @@ public struct FeedScreenContent: View {
         .onChange(of: searchQuery) { _ in
             viewModel.onSearchTextChanged(newQuery: searchQuery)
         }
-        .searchable(text: $searchQuery, prompt: "Search articles...")
+        .searchable(text: $searchQuery, prompt: "Search \(state?.searchQuery.lowercased() ?? "" )")
```

Comment:
Taking some liberty of putting this lowercased as it reads better in the Search box  `Search brexit,fintech` rather than the uppercased default?

Reply:
Also, due to the nullability of the `state`, an empty value needs to be provided, correct?

Reply:
The state should now never be null. We have a reasonable default which is “brexit,fintech” and we always expect it to be non-null. In the edge case where the search string is cleared we can again default it to “brexit,fintech”.

So I’d make state non-null for now and then we won’t have to worry about the UI layer. We should always be showing some search term 👍

Reply:
Also how about we put the searched string in quotes like Search “brexif,fintech”. I wonder if that’s possible?

Reply:
Now that I see this again, I think we should move the `searchable` modifier one level below to the `FeedState` struct since that is where we define our `AppNavigationView` and the whole screen content. In that case `@State private var searchQuery = ""` would also move in there.

Then, to notify the view-model, we can pass a callback function to `FeedState`, like `let onSearchChanged: (query: String) -> ()` which we'd call from `.onChange(of: searchQuery)`. Then, here in `FeedScreenContent` we can call our view-model the same way.

This achieves a better separation of concerns and allows it to keep the top-level `FeedScreenContent` separate as it's only used to show full screen blocking errors and setup the view-model.

Reply:
On placement:
- I agree overall, but I think `FeedState` is a bit of a misleading name for a `View` that wraps the `SectionedFeed`. I wonder if a rename is in place here given it's going to be the `View` that gets filtered by the `searchQuery`.

About nullability:
- I wonder why there's a nullcheck on the state and have to provide a default. As you say, 

> I don't expect it to be null since it has a reasonable default in the view-model

About putting the double quotes, I wonder how would it work with the String interpolation, I guess it would be a matter of escaping the double quotes.

Reply:
**On naming**

The current meaning is:
- FeedScreen: the screen that is pushed on the navigation stack when app starts
- FeedScreenContent: container for all possible states of the screen as well as the viewmodel
- FeedState, ErrorState etc: the individual states a screen can be into.

The reason we have an extra abstraction is because of the UI tests - we want to inject a viewmodel in there to simulate carious scenarios. Otherwise we can’t test the UI. The names are up for debate though 😂

So I think the logical spot for the search logic is within the FeedState since it’ll be shown if there aren’t errors. Then since the viewmodel is in the higher level we can pass a callback that’ll trigger when search text changes and we make the call to the viewmodel.

In compose this pattern is called [state hoistin](https://developer.android.com/jetpack/compose/state#state-hoisting)g and basically allows a composable to be stateless.

**On nullability**

We need an extra null check because FeedState handles loading and state could be null while loading. This is a personal choice here since we show the loading bar at the top of the navigation bar which lives as part of FeedState. Other scenarios might not handle the same way though 👍

Does this help a bit?

Reply:
Also, i imagine we could escape double quotes but I’m unsure. Not a huge deal tho